test/tests/logger-prod.test.js

```diff
@@ -0,0 +1,264 @@
+describe('graphql - query logging with sanitization in production', () => {
+  process.env.NODE_ENV = 'production'
+  const cds = require('@sap/cds/lib')
+  const path = require('path')
+  const { gql } = require('../util')
+  const util = require('util')
+
+  const PROJECT_PATH = path.join(__dirname, '../resources/bookshop-graphql')
+
+  const { axios, GET, POST, data } = cds.test(PROJECT_PATH)
+  // Prevent axios from throwing errors for non 2xx status codes
+  axios.defaults.validateStatus = false
+  data.autoReset(true)
+
+  const _format = e => util.formatWithOptions({ colors: false, depth: null }, ...(Array.isArray(e) ? e : [e]))
+
+  let _log
+
+  beforeAll(async () => {
+    await cds.deploy(path.join(PROJECT_PATH, 'srv'))
```

Comment:
@hm23 btw, since we were talking about this: This is a way to deploy to SQLite in-memory with `cds.test`, even if `NODE_ENV=production` is set.